fetchcode/__init__.py

```diff
@@ -0,0 +1,105 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from ftplib import FTP
+from mimetypes import MimeTypes
+import os
+import tempfile
+from urllib.parse import urlparse
+
+import requests
+
+
+class Response:
+    def __init__(self, location, content_type, size, url):
+        """
+        Represent the response from fetching a URL with:
+        - `location`: the absolute location of the files that was fetched
+        - `content_type`: content type of the file
+        - `size`: size of the retrieved content in bytes
+        - `url`: fetched URL
+        """
+        self.url = url
+        self.size = size
+        self.content_type = content_type
+        self.location = location
```

Comment:
What does the response class from `requests` look like? We may want to support a few more fields on our `Response` object and this may have some insights that we have overlooked.

Reply:
```['__attrs__', '__bool__', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__enter__', '__eq__', '__exit__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__nonzero__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_content', '_content_consumed', '_next', 'apparent_encoding', 'close', 'connection', 'content', 'cookies', 'elapsed', 'encoding', 'headers', 'history', 'is_permanent_redirect', 'is_redirect', 'iter_content', 'iter_lines', 'json', 'links', 'next', 'ok', 'raise_for_status', 'raw', 'reason', 'request', 'status_code', 'text', 'url']``` it looks like this

Reply:
@majurg ping, can you tell on which parameters we should start working :)

Reply:
@TG1999 We probably want to store `headers, status_code, encoding, content` and maybe others. 

We may want to subclass the `requests` Response object and just add on our own fields. @pombredanne Thoughts?

Reply:
Right now this `Response` class does not contain the actual data/content after fetching a uri. I am almost certain we want fetchcode utility to actually "fetch" whatever is at the uri.

Reply:
@TG1999 Are you always storing on disk at `location`? If so then we may not need the content... but in general the API of a response-like object has some method or attribute t get the content (see the Python requests package API)

Reply:
For now, the data is on disk at `location`. This is a detail I missed, so this should be resovled for now. We can think about adding this field/functionality in the future if needed.